src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1511,71 @@ private void WritePackageDependencies()
                 }
             }
 
+            private void WritePackageDependenciesBetweenPackages()
+            {
+                foreach(var target in _lockFile.Targets)
+                {
+                    GetPackageDependencies(target);
+                }
+            }
+
+            private void GetPackageDependencies(LockFileTarget target)
```

Comment:
This implementation is similar to the one in `ResolvePackageDependencies.RaiseLockFileTargets()` except that this implementation doesn't keep track of other output properties like `targetDefinitions`.